AnnotatoBackend/Sources/App/WebSocketControllers/WebSocketController.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+import Vapor
+import AnnotatoSharedLibrary
+import FluentKit
+
+class WebSocketController {
+    /// Each user has one websocket connection.
+    private static var openConnections: [String: WebSocket] = [:]
+    private static var logger = Logger(label: "WebSocketController")
+
+    static func handleIncomingConnection(req: Request, webSocket: WebSocket) {
+        guard let userId = try? ControllersUtil.getIdFromParamsAsString(request: req) else {
+            req.logger.info("Could not get user ID for request. Closing WebSocket connection...")
+            _ = webSocket.close()
+            return
+        }
+
+        req.logger.info("Client with ID: \(userId) connected!")
+        addToOpenConnections(userId: userId, webSocket: webSocket)
+
+        webSocket.onBinary(handleBinaryData(userId: userId, db: req.db))
+        webSocket.onText(handleTextData(userId: userId, db: req.db))
+
+        webSocket.onClose.whenComplete { _ in
+            req.logger.info("Client with ID: \(userId) disconnected!")
+            Self.handleClosedConnection(userId: userId, webSocket: webSocket)
+        }
+    }
+
+    static func sendToAllAppropriateClients<T: Codable>(
+        userId: String,
+        documentId: UUID,
+        db: Database,
+        message: T
+    ) async {
```

Comment:
It feels abit iffy for the controller here to decide which clients are the appropriate ones to send to. I think that responsibility should belong to the `AnnotationWebSocketController` or `DocumentWebSocketController`, and this function should just be 
```suggestion
    static func sendAll<T: Codable>(recipientIds: [String], message: T) async {
```
?